copylock vet errors in staging/src/k8s.io/component-base/metrics

Parent issue: https://github.com/kubernetes/kubernetes/issues/109197

```
❯ go vet ./staging/src/k8s.io/component-base/metrics/
# k8s.io/component-base/metrics
# [k8s.io/component-base/metrics]
staging/src/k8s.io/component-base/metrics/desc_test.go:160:26: call of reflect.DeepEqual copies lock value: k8s.io/component-base/metrics.Desc contains sync.RWMutex
staging/src/k8s.io/component-base/metrics/desc_test.go:160:34: call of reflect.DeepEqual copies lock value: k8s.io/component-base/metrics.Desc contains sync.RWMutex
```

I have some ideas for fixing it. Add an Equal function `func (d *Desc) Equal(other *Desc) bool` in `desc.go`, and use `if !descA.Equal(descB)` in `desc_test.go` to check. wdyt?

--- a/staging/src/k8s.io/component-base/metrics/desc_test.go
+++ b/staging/src/k8s.io/component-base/metrics/desc_test.go
@@ -17,7 +17,6 @@ limitations under the License.
 package metrics
 
 import (
-	"reflect"
 	"strings"
 	"testing"
 
@@ -150,15 +149,13 @@ func TestDescClearState(t *testing.T) {
 		tc := test
 		t.Run(tc.name, func(t *testing.T) {
 			descA := NewDesc(tc.fqName, tc.help, nil, nil, tc.stabilityLevel, tc.deprecatedVersion)
-			descB := NewDesc(tc.fqName, tc.help, nil, nil, tc.stabilityLevel, tc.deprecatedVersion)
-
 			descA.create(&currentVersion)
+
 			descA.ClearState()
 
-			// create
-			//nolint:govet // it's okay to compare sync.RWMutex, it's empty
-			if !reflect.DeepEqual(*descA, *descB) {
-				t.Fatal("descriptor state hasn't be cleaned up")
+			if descA.IsCreated() || descA.IsDeprecated() || descA.IsHidden() {
+				t.Fatalf("descriptor state hasn't be cleaned up; is created: %v, is deprecated: %v, is hidden: %v", descA.IsCreated(), descA.IsDeprecated(), descA.IsHidden())
+
 			}
 		})
 	}

